File: hackerrank/problem solving/algorithms/2 - implementation/30 - extra long factorials/extra_long_factorials_procedural.rb
# Source: https://www.hackerrank.com/challenges/extra-long-factorials/problem?isFullScreen=true

def main
    number = gets.to_i
    factorial = calculateFactorialOf number
    puts factorial
end

    def calculateFactorialOf number
        factorial = 1
        for i in 2..number
            factorial *= i
        end
        factorial
    end

main

File: hackerrank/problem solving/algorithms/2 - implementation/30 - extra long factorials/extra_long_factorials_procedural.java
// Source: https://www.hackerrank.com/challenges/extra-long-factorials/problem?isFullScreen=true

import java.util.Scanner;
import java.math.BigInteger;

public class Solution {
    private static Scanner _scanner;

    public static void main(String[] args) {
        _scanner = new Scanner(System.in);
        final int NUMBER = _scanner.nextInt();
        final BigInteger FACTORIAL = _calculateFactorialOf(NUMBER);
        System.out.println(FACTORIAL);
    }

        private static BigInteger _calculateFactorialOf(int number) {
            BigInteger factorial = BigInteger.ONE;
            for (int i = 2; i <= number; i++)
                factorial = factorial.multiply(BigInteger.valueOf(i));
            return factorial;
        }
}
